src/glue.rs.in

```diff
@@ -52,6 +61,22 @@ foreign_class!(class Fluvio {
 });
 
 
+pub struct ConsumerConfigWrapper {
+    wasm_module: Vec<u8>,
+}
+
+impl ConsumerConfigWrapper {
+    fn new_config_with_wasm_filter(file: &str) -> ConsumerConfigWrapper {
```

Comment:
I didn't catch it until I saw the stacktrace in CI but we should remove the `unwraps`. Due to the fact that [`std::fs::read`](https://doc.rust-lang.org/std/fs/fn.read.html) returns a `Result<Vec<u8>, std::io::Error>`, let's make this return something like `Result<ConsumerConfigWrapper, std::io::Error>`.

I'm a little hesitant to suggest it but given that `FluvioError` implements [`From<IoError>`](https://docs.rs/fluvio/0.13.1/fluvio/enum.FluvioError.html#method.from), this function could also just return a `Result<ConsumerConfigWrapper, FluvioError>` and have `raw_buffer.as_ref().unwrap()` be `raw_buffer.as_ref()?`. The `?` here is called the [question mark operator and is used to bubble up and handle various error types](https://doc.rust-lang.org/book/ch09-02-recoverable-errors-with-result.html). Basically, it's syntactic sugar that roughly expands to:

```rust
let good_val = match my_result_type { 
    Err(e) => return FluvioError::from(e)
    Ok(val) => val
};
```

The reason I'm a little hesitant about using the base `FluvioError` is that the conversions to a `FluvioError` (really just any rust error type) are done in that rust crate and are exposed upward. Something about having a library/crate that converts an io error downward a dependency's error rather than it's own error type _feels_ wrong. That being said, I think I've done it before.

Anyway, the `unwrap` causes this to segfault/panic rather than bubble up to the python as an exception. The Flapigen macros turns these error types into Strings (though, it can also return a typed error if we want) and raises them on the python side.

What do you think? We could do the simple change on the error type in this PR and then create an issue or have a follow up PR exposing a more expressive type to the Python side.

Reply:
I will implement your first suggestion and return, sorry about the delay, i was having a weekend away from the computer!

Reply:
I have implemented your suggestion, it now looks like this:

```rust
impl ConsumerConfigWrapper {
    fn new_config_with_wasm_filter(file: &str) -> Result<ConsumerConfigWrapper, std::io::Error> {
        let raw_buffer = match std::fs::read(file) {
            Ok(b) => b,
            Err(error) => return Err(error)
        };
        let mut encoder = GzEncoder::new(raw_buffer.as_slice(), Compression::default());
        let mut buffer = Vec::with_capacity(raw_buffer.len());
        match encoder.read_to_end(&mut buffer) {
            Ok(encoder) => encoder,
            Err(error) => return Err(error)
        };
        Ok(ConsumerConfigWrapper {
            wasm_module: buffer,
        })
    }
}
```
The only bit i was unsure on, was in the `stream_with_config` method, i was unsure of how to handle the error from the result so i settled on this:

```rust
pub fn stream_with_config(
        consumer: &PartitionConsumer,
        offset: &Offset,
        wasm_module_path: &str
    ) -> Result<PartitionConsumerStream, FluvioError> {
        let config_wrapper = match ConsumerConfigWrapper::new_config_with_wasm_filter(wasm_module_path) {
            Ok(config) => config,
            Err(error) => return Err(FluvioError::Other(error.to_string()))
        };
        let mut builder = ConsumerConfig::builder();
        builder.smartmodule(Some(SmartModuleInvocation {
            wasm: SmartModuleInvocationWasm::AdHoc(config_wrapper.wasm_module),
            kind: SmartModuleKind::Filter,
            params: Default::default()
        }));
        let config = builder.build().map_err(|err| FluvioError::Other(err.to_string()))?;
        run_block_on(consumer.stream_with_config(offset.clone(), config)).map(|stream| PartitionConsumerStream { inner: Box::pin(stream) })
    }
```